.github/workflows/tests.yml

```diff
@@ -174,19 +174,19 @@ jobs:
         uses: Homebrew/actions/setup-homebrew@master
 
       - name: Build Docker image
-        run: docker build -t brew --build-arg=version=16.04 .
+        run: docker build -t brew --build-arg=version=22.04 .
 
       - name: Deploy the Docker image to GitHub Packages and Docker Hub
         if: github.ref == 'refs/heads/master'
         run: |
           echo ${{secrets.HOMEBREW_BREW_GITHUB_PACKAGES_TOKEN}} |
             docker login ghcr.io -u BrewTestBot --password-stdin
-          docker tag brew "ghcr.io/homebrew/ubuntu16.04:master"
-          docker push "ghcr.io/homebrew/ubuntu16.04:master"
+          docker tag brew "ghcr.io/homebrew/ubuntu22.04:master"
+          docker push "ghcr.io/homebrew/ubuntu22.04:master"
           echo ${{secrets.HOMEBREW_BREW_DOCKER_TOKEN}} |
             docker login -u brewtestbot --password-stdin
-          docker tag brew "homebrew/ubuntu16.04:master"
-          docker push "homebrew/ubuntu16.04:master"
+          docker tag brew "homebrew/ubuntu22.04:master"
+          docker push "homebrew/ubuntu22.04:master"
```

Comment:
How do we handle compatibility for those currently using `homebrew/ubuntu16.04:master`?

Update both? Alias `master` to `latest`?

Reply:
@Bo98 what are the compatibility issues here?

Reply:
Existing workflows which reference `homebrew/ubuntu16.04:master`, which would suddenly stop being updated with this change (but `homebrew/ubuntu16.04:latest` still will).

Reply:
@Bo98 Ah, I see. Yeh, feels like it would be nice to push both tags.

Reply:
We push `homebrew/ubuntu16.04:latest` with each stable release of Homebrew. I'd like to permanently remove `homebrew/ubuntu16.04:master` after we updated to `homebrew/ubuntu22.04:master`.

Reply:
The problem is communicating that to third party taps (e.g. https://github.com/brewsci/homebrew-num/blob/master/.github/workflows/upload-bottles.yml). We should ideally provide some deprecation messaging.

Ignoring `master` for a moment, what's your thoughts on the future of the Ubuntu 16.04 image in general?

Reply:
> The problem is communicating that to third party taps (e.g. https://github.com/brewsci/homebrew-num/blob/master/.github/workflows/upload-bottles.yml). We should ideally provide some deprecation messaging.

Agreed.

Reply:
@sjackman I consider this unresolved comment to be blocking the 3.6.0 release, let's figure out a backwards compatibility/deprecation story here please.

Reply:
If there's a desire to deprecate (not clear whether that's just `master` or `ubuntu16.04` as a whole), we could implemented a system in brew where we can insert a gitignored file into the Docker image that will add a deprecation message to each invokation of `brew` (and as a login shell message too if you'd like).

This can be tested out with `Homebrew/debian7`, which we don't need to provide a deprecation notice for but we can use it as a test.

Reply:
I think we should issue a deprecation notice for `ubuntu16.04`, and one for `ubuntu18.04` when it goes EOL next year as well.  Testing on `debian7` makes sense, because we no longer use it for anything.

Reply:
I wonder whether we even need to deprecate these rather than (as @sjackman has proposed below and I'd be favour of) just make them throw an error message with `master` or `latest`.